Library/Homebrew/test/cli_parser_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "../cli_parser"
+
+describe Homebrew::CLI::Parser do
+  describe "test switch options" do
+    subject(:parser) {
+      described_class.new do
+        switch "--verbose",      "Flag for verbosity"
+        switch "--more-verbose", "Flag for higher verbosity"
+      end
+    }
+    it "with a valid option passed" do
+      args = parser.parse(["--verbose"])
+      expect(args.verbose?).to be true
+    end
+
+    it "with valid option and unnamed args passed" do
+      args = %w[--verbose unnamed args]
+      parser.parse(args)
+      expect(args).to eq %w[unnamed args]
+    end
+
+    it "with a single option passed" do
+      args = parser.parse(["--verbose"])
+      expect(args.verbose?).to be true
```

Comment:
I prefer `expect(args).to be_verbose`, but if you like it this way I think it's fine.